app/src/main/java/com/ncnf/views/activities/group/FriendsTrackerActivity.java

```diff
@@ -277,6 +341,36 @@ private boolean isLocationServiceRunning() {
         return false;
     }
 
+    public void getImagesForClusters() {
+
+        for(int i = 0; i < friendsUUID.size(); ++i) {
+            fileStore.setContext(getApplicationContext());
+            fileStore.setPath(USER_IMAGE_PATH, friendsUUID.get(i) + ".jpg");
+            CompletableFuture<byte[]> image = fileStore.download();
+
+            int finalI = i;
+            image.thenAccept(data -> {
+                Bitmap bitmap = BitmapFactory.decodeByteArray(data, 0, data.length);
+                images.put(friendsUUID.get(finalI), bitmap);
+                bitmapSetChanged();
+            }).exceptionally(e -> {
+                images.put(friendsUUID.get(finalI), decodeResource(getApplicationContext().getResources(), R.drawable.default_profile_picture));
+                bitmapSetChanged();
+                return null;
+            });
+
+        }
+
+    }
+
+    private void bitmapSetChanged() {
+        if(images.keySet().size() == friendsUUID.size()) {
+            if(markers.keySet().size() == friendsUUID.size()) {
```

Comment:
This is really minor, but wouldn't it be cleaner to just have if (a & b){} instead of if (a){if (b){}}?